File: algo-library/python/megruBisect.py
def megru_bisect(ng,ok):
  def judge(mid):
  
  while abs(ok-ng)>1:
    mid = (ok+ng)//2
    if judge(mid): # 条件を満たすなら
      ok = mid
    else:
      ng = mid

  return ng,ok

File: algorithm_library/cpp-from-py/techniques/MegruBisect.cpp
#include <bits/stdc++.h>
using namespace std;

template <class Judge>
pair<long long, long long> megru_bisect(long long ng, long long ok, Judge judge) {
    while (abs(ok - ng) > 1) {
        long long mid = (ok + ng) / 2;
        if (judge(mid)) ok = mid;
        else ng = mid;
    }
    return {ng, ok};
}
